src/java/org/apache/cassandra/service/accord/serializers/AwaitSerializers.java

```diff
@@ -114,8 +114,8 @@ public long serializedSize(A await, int version)
         }
     };
 
-    public static final IVersionedSerializer<AwaitOk> syncReply = new EnumSerializer<>(AwaitOk.class);
-    public static final IVersionedSerializer<RecoverAwaitOk> recoverReply = new EnumSerializer<>(RecoverAwaitOk.class);
+    public static final IVersionedSerializer<AwaitOk> syncReply = EncodeAsVInt32.of(AwaitOk.class);
+    public static final IVersionedSerializer<RecoverAwaitOk> recoverReply = EncodeAsVInt32.of(RecoverAwaitOk.class);
```

Comment:
reviewing this as it conflicts with https://github.com/apache/cassandra/pull/3954/

why this change? im am not against it, just trying to understand the motivations.

Reply:
Merging multiple different serialisers into a single one. We had SmallEnumSerializer, EnumSerializer, wrapped both of them in nullable unnecessarily, and would have needed another new one for Known (which isn't an enum, but is easily stored as an integer, and was the motivation). So, this replaced about five logically distinct serializers.